### PR TITLE
Comm Player Manager

### PR DESCRIPTION
**Pull request by adrienntindall**
Not completely finished but I'm plateauing and Lhea is cooking

### CODE REVIEW

File: include/struct_defs/struct_02057B48.h [hunk 1]
```diff
@@ -2,12 +2,12 @@
 #define POKEPLATINUM_STRUCT_02057B48_H
 
 typedef struct {
```
Comment (lhearachel) [inline include/struct_defs/struct_02057B48.h:4]: `typedef struct CommPlayerLocation {`

File: src/comm_player_manager.c [hunk 2]
```diff
@@ -0,0 +1,1825 @@
...
+
+    CommSys_EnableSendMovementData();
+    CommPlayer_InitPersonal();
+
+    return TRUE;
+}
+
+void CommPlayerMan_Reset (void)
+{
+    int netId;
```
Comment (lhearachel) [inline src/comm_player_manager.c:120]: As above

File: src/comm_player_manager.c [hunk 3]
```diff
@@ -0,0 +1,1825 @@
...
+
+    if ((keys & PAD_KEY_UP)) {
+        return 0;
+    }
+
+    if ((keys & PAD_KEY_DOWN)) {
+        return 1;
+    }
+
+    return(-1);
```
Comment (lhearachel) [inline src/comm_player_manager.c:654]: Unnecessary parens

**Review by lhearachel — COMMENTED** [review]
Going lax on the style nitpicks here, since they'll be fixed by the clang-format PR.

File: src/comm_player_manager.c [hunk 4]
```diff
@@ -0,0 +1,1825 @@
...
+    sCommPlayerManager->unk_2BB = TRUE;
+    sCommPlayerManager->isResetting = FALSE;
+
+    CommPlayer_InitPersonal();
+    CommPlayer_SendPos(TRUE);
+}
+
+void CommPlayerMan_Delete (BOOL deletePlayerData)
+{
+    int netId, netJd;
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/comm_player_manager.c:154]: netId and netJd usually get used as iterators, and I thought it'd be funny

File: include/struct_defs/struct_02057B48.h [hunk 5]
```diff
@@ -2,12 +2,12 @@
...
-    u16 unk_00;
-    u16 unk_02;
-    s8 unk_04;
-    u8 unk_05;
+    u16 x;
+    u16 z;
```
Comment (cbt6) [inline include/struct_defs/struct_02057B48.h:6]:
```suggestion
    u16 y;
```

Comment (adrienntindall) [reply include/struct_defs/struct_02057B48.h:6]: as we have established in the hgss repo, we use the cringe z-up notation due to the orientation of (x, y, z) vs (x, z, y) causing the vector math to be different

Comment (lhearachel) [reply include/struct_defs/struct_02057B48.h:6]: As cringe as it is, it's just how the DS assumes planes are rendered, to my knowledge. Vector math in the Pokemon games, at least, is contingent on the player running along the XZ plane (and Platinum specifically makes this important with the Distortion World).

File: include/comm_player_manager.h [hunk 6]
```diff
@@ -4,27 +4,27 @@
...
-void sub_02057764(void);
-void sub_020578B0(void);
-void sub_020578DC(void);
-void sub_020579BC(int param0);
-void sub_02057A94(BOOL param0, int param1, int param2);
-void sub_02057AE4(BOOL param0);
-void sub_02057B14(BOOL param0);
-u32 sub_02057C84(void);
-void sub_02057DB8(u8 param0, BOOL param1, BOOL param2);
+CommPlayerManager * CommPlayerData_Get(void);
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply include/comm_player_manager.h:7]: Ah yeah artifact from a previous version of the struct name

File: src/comm_player_manager.c [hunk 7]
```diff
@@ -0,0 +1,1825 @@
...
+static CommPlayerManager * sCommPlayerManager = NULL;
+
+CommPlayerManager * CommPlayerData_Get (void)
+{
+    return sCommPlayerManager;
+}
+
+BOOL CommPlayerManager_Init (void * dest, FieldSystem * fieldSys, BOOL isUnderground)
+{
+    int netId;
```
[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/comm_player_manager.c:71]: not sure what you mean?

[the remark this answers is not in the document]
Comment (lhearachel) [reply src/comm_player_manager.c:70]: This variable isn't used outside of the `for` loop below:

```c
for (netId = 0; netId < MAX_CONNECTED_PLAYERS; netId++) {
    // snip
}
```

My question is if you can limit the declaration scope to the loop itself and have it still match:

```c
for (int netId = 0; netId < MAX_CONNECTED_PLAYERS; netId++) {
    // snip
}
```

[the remark this answers is not in the document]
Comment (adrienntindall) [reply src/comm_player_manager.c]: Looks like a lot were able to be inlined like that yeah, thanks for the clarification

**Review by lhearachel — APPROVED** [review]

**Pull request merged**